tools/schemacode/bidsschematools/render.py

```diff
@@ -287,9 +287,29 @@ def make_filename_template(schema, n_dupes_to_combine=6, **kwargs):
     paragraph = paragraph.rstrip()
     codeblock = "Template:\n```Text\n" + paragraph + "\n```"
     codeblock = codeblock.expandtabs(4)
+    codeblock = append_filename_template_legend(codeblock)
     return codeblock
 
 
+def append_filename_template_legend(text=None):
+
+    if text is None:
+        return
+
+    text += """
+
+**Legend:**
+
+- Filename entities or folders between square brackets (for example `[_ses-<label>]`) are OPTIONAL.
+- See the glossary for a definition of [`label`](../99-appendices/14-glossary.md#label)
+  and [`index`](../99-appendices/14-glossary.md#index).
+- See the [entity page](../99-appendices/09-entities.md) for a definition of each entity.
```

Comment:
Given this is now showing up multiple times in some pages (namely MRI), I wonder if making it collapseable would be better:

```suggestion
<details>
<summary><strong>Legend:</strong></summary>

- Filename entities or folders between square brackets (for example `[_ses-<label>]`) are OPTIONAL.
- See the glossary for a definition of [`label`](../99-appendices/14-glossary.md#label)
  and [`index`](../99-appendices/14-glossary.md#index).
- See the [entity page](../99-appendices/09-entities.md) for a definition of each entity.

</details>
```

I have obviously not tested if this would

Reply:
PDF rendering would have to adapted too, no?

Reply:
commit this.
let's see how the whole thing behaves

Reply:
Hum... Not quite.

![Screenshot from 2022-08-18 09-57-46](https://user-images.githubusercontent.com/6961185/185342123-8f622d17-2fe8-40f6-a461-d0c5b3c365e9.png)

Reply:
It looks like the PDF doesn't show the legend at all, which I think is totally fine. I'm not sure why the legend no longer respects Markdown though.